st2api/st2api/controllers/v1/actions.py

```diff
@@ -78,8 +78,13 @@ def __init__(self, *args, **kwargs):
 
     @request_user_has_permission(permission_type=PermissionType.ACTION_LIST)
     @jsexpose()
-    def get_all(self, **kwargs):
-        return super(ActionsController, self)._get_all(**kwargs)
+    def get_all(self, exclude_attributes=None, **kwargs):
+        if exclude_attributes:
```

Comment:
I'd rather prefer us to choose the list of fields we need instead of the one we exclude. Reason being, client developer knows for sure what kind of data he wants to show. Any field we add to the model on top of that will make response bigger and will require client developer to add it to exclusion list. By using `.only()` instead of `.exclude()` we can ensure performance won't degrade over time. Saying that, yeah, the refactoring is needed.